scripts/release-notes/index.js

```diff
@@ -49,11 +56,138 @@ const PREFIXES_TO_IGNORE = [
   'Type: Investigation',
 ];
 
-const getMilestone = async () => {
-  const response = await octokit.rest.issues.listMilestones({ owner: OWNER, repo: REPO_NAME });
-  return response.data.find(milestone => milestone.title === MILESTONE_NAME);
+const getRepoQueryString = query => `{ repository(owner: "${OWNER}", name: "${REPO_NAME}") { ${query} } }`;
+
+const queryRepo = query => octokit.graphql(getRepoQueryString(query));
+
+const queryRepoPaginated = query => octokit.graphql
+  .paginate(`query paginate($cursor: String) ${getRepoQueryString(query)}`);
+
+const getLatestReleaseName = async () => queryRepo(
+  `releases(first: 1) {
+        edges { node { tagName } }
+      }`
+).then(({ repository }) => repository.releases.edges[0].node.tagName);
+
+const getMilestoneBranch = async () => {
+  const milestoneBranch = [...MILESTONE_NAME.split('.').slice(0, -1), 'x'].join('.');
+  const branchExists = await queryRepo(
+    `ref(qualifiedName: "refs/heads/${milestoneBranch}") {
+        target { oid }
+      }`
+  ).then(({ repository }) => repository.ref);
+  if (branchExists) {
+    return milestoneBranch;
+  }
+
+  // Fall back to default branch if milestone branch doesn't exist. This might be useful when preparing for a release
+  // before actually creating the release branch.
+  return queryRepo(
+    `defaultBranchRef { name }`
+  ).then(({ repository }) => repository.defaultBranchRef.name);
+};
+
+// This query calculates the "commits for the release" by comparing the commit history of the latest release with the
+// commit history of the milestone branch and keeping only the commits that are unique to the milestone branch.
+const getCommitsForRelease = async (release, milestoneBranch) => queryRepoPaginated(
+  `ref(qualifiedName: "${release}") {
+      compare(headRef: "${milestoneBranch}") {
+        commits(first: 100, after: $cursor) {
```

Comment:
That's a much better way to get commits for a given release than what I had :sunglasses: :tada: